src/iterable_recursive_open_struct.rb

```diff
@@ -0,0 +1,26 @@
+
+module Metalware
+  class IterableRecursiveOpenStruct < RecursiveOpenStruct
+    def each(&block)
+      convert_hash_values_to_own_class.each(&block)
+    end
+
+    def each=(*args)
+      raise IterableRecursiveOpenStructPropertyError,
+        "Cannot set property 'each', reserved to use for iteration"
+    end
+
+    private
+
+    def convert_hash_values_to_own_class
```

Comment:
I know it is unlikely this will happen as the method is a long name. But we should preface convert_hash with a _. This would make it `_convert_hash_values_to_own_class`. Alternatively set a `convert_hash_values_to_own_class=` method that explicitly prohibits it from being overloaded.